python/ray/data/_internal/datasource/parquet_datasource.py

```diff
@@ -776,9 +778,25 @@ def get_parquet_dataset(paths, filesystem, dataset_kwargs):
             **dataset_kwargs,
             filesystem=filesystem,
         )
+    except TypeError:
+        # Fallback: resolve filesystem locally in the worker
+        try:
+            resolved_paths, resolved_filesystem = _resolve_paths_and_filesystem(
+                paths, filesystem=None
+            )
+            resolved_filesystem = RetryingPyFileSystem.wrap(
+                resolved_filesystem,
+                retryable_errors=DataContext.get_current().retried_io_errors,
+            )
+            dataset = pq.ParquetDataset(
+                resolved_paths,
+                **dataset_kwargs,
+                filesystem=resolved_filesystem,
+            )
+        except OSError as os_e:
```

Comment:
Ok, so after some research i think the reason why it's failing is that PA 14.0 isn't handling PyFileSystems correctly, and your fix actually won't work since you're rewrapping it before passing to ParquetDataset

This doesn't seem to be a problem in PA 21.0 though.

So i think the proper fix would be to unwrap our RetryignPyFileSystem when passing it to get_parquet_dataset for PA < 14.0